.github/workflows/app_build.yml

```diff
@@ -0,0 +1,304 @@
+        echo "::set-env name=PLATFORM_NAME::${{ matrix.PLATFORM_NAME }}"
+        echo "::set-env name=CPACK_BINARY_DEB::${{ matrix.CPACK_BINARY_DEB }}"
+        echo "::set-env name=CPACK_BINARY_IFW::${{ matrix.CPACK_BINARY_IFW }}"
+        echo "::set-env name=BINARY_EXT::${{ matrix.BINARY_EXT }}"
+
+        if [ "$RUNNER_OS" == "Linux" ]; then
+          echo "Install needed system dependencies for OPENGL (due to Qt) for Linux"
+          sudo apt install -y mesa-common-dev libglu1-mesa-dev patchelf
+        elif [ "$RUNNER_OS" == "Windows" ]; then
+          curl -L -O https://download.qt.io/official_releases/qt-installer-framework/3.2.2/QtInstallerFramework-win-x86.exe
+          ./QtInstallerFramework-win-x86.exe --verbose --script ./ci/install_script_qtifw.qs
+          echo "::add-path::C:/Qt/QtIFW-3.2.2/bin"
+        elif [ "$RUNNER_OS" == "macOS" ]; then
+          echo "::set-env name=MACOSX_DEPLOYMENT_TARGET::${{ matrix.MACOSX_DEPLOYMENT_TARGET }}"
+          echo "::set-env name=SDKROOT::${{ matrix.SDKROOT }}"
+          echo "::set-env name=CMAKE_MACOSX_DEPLOYMENT_TARGET::'-DCMAKE_OSX_DEPLOYMENT_TARGET=$MACOSX_DEPLOYMENT_TARGET'"
+
+          curl -L -O https://download.qt.io/official_releases/qt-installer-framework/3.2.2/QtInstallerFramework-mac-x64.dmg
+          hdiutil attach -mountpoint ./qtfiw_installer QtInstallerFramework-mac-x64.dmg
+          echo "ls ./qtfiw_installer"
+          ls ./qtfiw_installer
+          echo "ls ./qtfiw_installer/QtInstallerFramework-mac-x64.app/"
+          echo "ls ./qtfiw_installer/QtInstallerFramework-mac-x64.app/Contents/"
+          echo "ls ./qtfiw_installer/QtInstallerFramework-mac-x64.app/Contents/MacOS"
+          echo "ls ./qtfiw_installer/QtInstallerFramework-mac-x64.app/Contents/Resources"
+          echo "ls ./qtfiw_installer/QtInstallerFramework-mac-x64.app/Contents/Frameworks"
+
+          sudo ./qtfiw_installer/QtInstallerFramework-mac-x64.app/Contents/MacOS/QtInstallerFramework-mac-x64 --verbose --script ./ci/install_script_qtifw.qs
+          echo "::add-path::~/Qt/QtIFW-3.2.2/bin/"
+        fi;
+
+    # TODO: cache the QtIFW folder too
+
+    - name: Cache OpenStudio tar.gz
+      # To avoid downloading the SDK all the time, we try to cache it
+      id: cacheossdk
+      uses: actions/cache@v2
+      with:
+        path: OpenStudio-${{ env.OS_SDK_VERSION }}
+        key: OpenStudio-SDK-${{ runner.os }}-${{ env.OS_SDK_VERSION }}
+
+    - name: Download the OpenStudio installer
+      # If the SDK wasn't found in the cache
+      if: steps.cacheossdk.outputs.cache-hit != 'true'
+      # Use a bash shell so we can use the same syntax for environment variable
+      # access regardless of the host operating system
+      # Note: wget not available on windows, even in bash, so use curl
+      shell: bash
+      run: |
+        curl --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' https://api.github.com/repos/NREL/OpenStudio/releases/tags/v$OS_SDK_VERSION > os_tag.json
```

Comment:
We probably need a way to point this at a branch or even a fork of the SDK

Reply:
A fork would proabably be too burdensome, because we'd have to actually get github actions to **build** the SDK before building the app. I doubt we want that.

A PR to OpenStudio is _doable_, since NREL's CI (jenkins) will build and upload installers that we could use, but that's going to be a lot of work to get _just_ right:

* you need an easy way to be able to specify which SDK PR to go reference... perhaps in the OpenStudioApplication PR message. I've thought about it for Jenkins a few months back - I would create a PR template so it could be reproducible
* then locate the right installer from that (and that is harder)